### ISSUE TITLE
#Distributed Tensorflow

### ISSUE THREAD

**Issue by chinhncse62121** [tensorflow/tensorflow#21115]
* Overview: I'm struggling  with "Distributed Tensorflow" and the error is "Could not satisfy explicit device specification '/job:worker/task:0' because the node was colocated with a group of nodes that required incompatible device '/job:ps/task:0'". I also try to research on StackOverflow,.. but it doesn't help much. 

The code I used to work with "Distributed Tensorflow":  https://imgur.com/Wna7Yhz

Thank you so much

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by tensorflowbutler** [comment]
Nagging Assignee @shivaniag: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by shivaniag** [comment]
@guptapriya could you take a look.

**Comment by guptapriya** [comment]
It seems like you're using the approach described here: https://www.tensorflow.org/deploy/distributed. 
Unfortunately, this approach doesn't work with keras fit, as far as I know. cc @anj-s - who can perhaps add more. 

If you'd like to run your training in keras and distribute on multiple GPUs, you can try using DistributionStrategy instead, which is now supported in keras fit. See example here for now: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/distribute/python/keras_test.py#L244
We have more documentation coming soon.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @guptapriya: It has been 20 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by guptapriya** [comment]
Closing as there is nothing to be done from our side currently (apart from supporting multi node with keras directly which we are actively working on)
